Pseudo_experiments/nbconverted/generate_E_GEOD_51409_template_experiment.py

```diff
@@ -0,0 +1,310 @@
+local_dir = params["local_dir"]
+
+
+# In[4]:
+
+
+# Input files
+base_dir = os.path.abspath(
+  os.path.join(
+      os.getcwd(), "../"))    # base dir on repo
+
+# Load experiment id file
+# Contains ALL experiment ids
+experiment_ids_file = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    "experiment_ids.txt")
+
+normalized_data_file = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "input",
+    "train_set_normalized.pcl")
+
+original_data_file = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    "Pa_compendium_02.22.2014.pcl")
+
+mapping_file = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    "sample_annotations.tsv")
+
+
+# ## Generate simulated data with labels
+# 
+# Simulate a compendia by experiment and label each new sample with the experiment id that it originated from
+
+# In[5]:
+
+
+# Load experiment id file
+# Contains ALL experiment ids
+base_dir = os.path.abspath(
+  os.path.join(
+      os.getcwd(), "../"))    # base dir on repo
+
+experiment_ids_file = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    "experiment_ids.txt")
+
+
+# In[6]:
+
+
+# Generate simulated data
+# Generate simulated data
+simulated_labeled_data_file = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    "simulated_data_labeled.txt.xz")
+if not Path(simulated_labeled_data_file).exists():
+    generate_labeled_data.simulate_compendium_labeled(experiment_ids_file, 
+                                                      num_simulated_experiments,
+                                                      normalized_data_file,
+                                                      NN_architecture,
+                                                      dataset_name,
+                                                      local_dir,
+                                                      base_dir)
+
+
+# ## Process data
+
+# In[7]:
+
+
+# Load simulated data
+simulated_data_file = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    "simulated_data_labeled.txt.xz")
+
+
+# In[8]:
+
+
+# Read data
+original_data = pd.read_table(
+    original_data_file,
+    header=0,
+    sep='\t',
+    index_col=0).T
+
+simulated_data = pd.read_table(
+    simulated_data_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+
+print(original_data.shape)
+print(simulated_data.shape)
+
+
+# In[9]:
+
+
+original_data.head(5)
+
+
+# In[10]:
+
+
+simulated_data.head(5)
+
+
+# In[11]:
+
+
+# 0-1 normalize per gene
+scaler = preprocessing.MinMaxScaler()
+original_data_scaled = scaler.fit_transform(original_data)
+original_data_scaled_df = pd.DataFrame(original_data_scaled,
+                                columns=original_data.columns,
+                                index=original_data.index)
+
+original_data_scaled_df.head(5)
+
+
+# In[12]:
+
+
+# Re-scale simulated data back into the same range as the original data
```

Comment:
Just confused, why are you scaling the original data and then un-scaling the simulated data? I got a little turned around in this part.

Reply:
This is definitely confusing. I will add a comment about this. 

Originally the expression data was 0-1 normalized for use in training the VAE, however when we started performing followup DE analyses and pathway analyses we found that the normalized data had reduced variance that resulted in an inconsistency between the number of DEGs found compared to the publication. So we are re-scaling/un-scaling our normalized data to be in the original range of data.